qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/ClusteringRule.java

```diff
@@ -208,16 +207,40 @@ protected void before() throws IOException {
 
   @Override
   protected void after() {
+    final var latch = new CountDownLatch(brokers.size());
+    brokers
+        .values()
+        .parallelStream()
+        .forEach(
+            broker -> {
+              broker.close();
+              latch.countDown();
+            });
 
-    brokers.values().parallelStream().forEach(Broker::close);
+    // wait until the brokers are shutdown before deleting their data, otherwise exceptions will be
+    // thrown and cause developers to waste time figuring out that they aren't production bugs
+    try {
+      latch.await(10, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      LOG.warn("Interrupted while waiting for {} brokers to close", brokers.size());
+    }
+
+    for (final var base : brokerBases.entrySet()) {
+      try {
+        FileUtil.deleteFolder(base.getValue().getAbsolutePath());
```

Comment:
Who did this before?

Reply:
Ah I see it is done in the closeables. But why not anymore? Brokers should be closed now right?

Reply:
I'm actually not sure why we keep track of the folders here and also in the closeables to be honest. My issue with closeables is basically that keeping track of resources differently (i.e. one is the resource itself, another is a closure capturing it) led to out of memory errors for me (since we removed things from `brokerBase` and `brokers` but not closeables).

So like I said, I'm actually not sure what the gain of closeables is for the brokers if we keep track of those values in another map anyway :sweat_smile:

Reply:
Furthermore if we use a temp folder rule, this should cleanup by himself right?